src/validator/BalanceValidator.cpp

```diff
@@ -19,13 +19,20 @@ Status BalanceValidator::toPlan() {
     case BalanceSentence::SubType::kLeader:
         current = BalanceLeaders::make(qctx_, nullptr);
         break;
-    case BalanceSentence::SubType::kData:
+    case BalanceSentence::SubType::kData: {
+        auto hosts = sentence->hostDel() == nullptr ? std::vector<HostAddr>()
+                                                    : sentence->hostDel()->hosts();
+        if (!hosts.empty()) {
+            auto it = std::unique(hosts.begin(), hosts.end());
+            if (it != hosts.end()) {
+                return Status::SemanticError("Host have duplicated");
```

Comment:
I think it no need to return an error for the user.

Reply:
> I think it no need to return an error for the user.

Whether to indicate an error depends on the impact the duplication may introduce.